scalding-core/src/main/scala/com/twitter/scalding/RichPipe.scala

```diff
@@ -36,7 +36,23 @@ object RichPipe extends java.io.Serializable {
 
   def getNextName: String = "_pipe_" + nextPipe.incrementAndGet.toString
 
-  def assignName(p: Pipe): Pipe = new Pipe(getNextName, p)
+  private val FormerNameBitLength = 8
+  private val FormerNameBitPattern = "^_pipe_([0-9]+)".r
+  private def getFormerNameBit(p: Pipe): String =
```

Comment:
meta: indeed (yes, riding on a PR isn't very good form). I'll move that tomorrow morning (UTC+2)
content: indeed too (pattern matching on extractors beyond Some(_) isn't yet on my toolbox' top drawer), will do it that way in the other PR.

Thanks for the feedback!